gems/aws-sdk-s3/spec/object/download_file_spec.rb

```diff
@@ -174,6 +183,53 @@ module S3
               ":chunk_size shouldn't exceed total file size."
             )
         end
+
+        context 'file size is chunk size boundary' do
+          before :each do
+            allow(client).to receive(:head_object).with(
```

Comment:
Why not use stub_responses for this? or does it need argument matching?

Reply:
Other tests needed it because the test files were re-used. I'm just following existing pattern.